### DISCUSSION TITLE
Bug using class: 'BaseSettings'

### DISCUSSION THREAD

**Discussion by chitohugo** [ets-labs/python-dependency-injector#831] [General]
Guys, I'm having a problem with the library, when using a Pydantic scheme and inheriting from the class: 'BaseSettings' I get an exception:

`pydantic.errors.PydanticImportError: `BaseSettings` has been moved to the `pydantic-settings` package. See https://docs.pydantic.dev/2.10/migration/#basesettings-has-moved-to-pydantic-settings for more details.`

Looking for what caused the exception, I found that Pydantic moved said class to another library.

I would appreciate it if you could replicate the error and make a fix. 

Thank you!
![image](https://github.com/user-attachments/assets/36430732-2c9e-408b-8c99-da0e40f393c5)

**Comment by ZipFile** [comment]
Fixed in #832, already available in the latest release.
